python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,1089 @@
+        # By default this will never happen since posFlux = negFlux.
+        passesFluxPos = (abs(measRecord[self.posFluxKey]) /
+                         (measRecord[self.fluxKey]*2.)) < self.config.maxFluxRatio
+        passesFluxPos &= (abs(measRecord[self.posFluxKey]) >= 1.0)
+        passesFluxNeg = (abs(measRecord[self.negFluxKey]) /
+                         (measRecord[self.fluxKey]*2.)) < self.config.maxFluxRatio
+        passesFluxNeg &= (abs(measRecord[self.negFluxKey]) >= 1.0)
+        allPass = (passesSn and passesFluxPos and passesFluxNeg)  # and passesChi2)
+
+        # Third, is it a good fit (chi2dof < 1)?
+        # Use scipy's chi2 cumulative distrib to estimate significance
+        # This doesn't really work since I don't trust the values in the variance plane (which
+        #   affects the least-sq weights, which affects the resulting chi2).
+        # But I'm going to keep this here for future use.
+        if False:
+            from scipy.stats import chi2
+            ndof = result.psfFitChi2 / measRecord[self.chi2dofKey]
+            significance = chi2.cdf(result.psfFitChi2, ndof)
+            passesChi2 = significance < self.config.maxChi2DoF
+            allPass = allPass and passesChi2
+
+        if allPass:  # Note cannot pass `allPass` into the `measRecord.set()` call below...?
+            measRecord.set(self.classificationFlagKey, True)
+        else:
+            measRecord.set(self.classificationFlagKey, False)
+
+    # TBD: need to catch more exceptions, set correct flags.
+    def fail(self, measRecord, error=None):
+        measRecord.set(self.flagKey, True)
+        if error is not None:
+            if error.getFlagBit() == self.FAILURE_EDGE:
+                measRecord.set(self.edgeFlagKey, True)
+            if error.getFlagBit() == self.FAILURE_FIT:
+                measRecord.set(self.flagKey, True)
+            if error.getFlagBit() == self.FAILURE_NOT_DIPOLE:
+                measRecord.set(self.flagKey, True)
+
+
+class DipoleFitAlgorithm(object):
+    """
+    Lightweight class containing methods for fitting dipoles in diffims, used by DipoleFitPlugin.
+    This code is documented in DMTN-007.
+    """
+    # Below is a (somewhat incomplete) list of improvements
+    # that would be worth investigating, given the time:
+
+    # 1. Initial fast test whether a background gradient needs to be fit
+    # 2. Initial fast estimate of background gradient(s) params -- perhaps using numpy.lstsq
+    # 3. better estimate for staring flux when there's a strong gradient
+    # 4. evaluate necessity for separate parameters for pos- and neg- images
+    # 5. only fit background OUTSIDE footprint and dipole params INSIDE footprint?
+    # 6. requires a new package `lmfit` -- investiate others? (astropy/scipy/iminuit?)
+    # 7. account for PSFs that vary across the exposures
+
+    # Only import lmfit if someone wants to use the new DipoleFitAlgorithm.
```

Comment:
You're right, just as @yalsayyad also said. I've moved the lmfit import into the fitDipole() method.